### PR TITLE
Refactor code

### PR DESCRIPTION
**Pull request by ElliottKasoar**
Mainly driven by the CLI module becoming far too large, but also removes some code duplication e.g. writing out the summary file. (Split from #119)

### CODE REVIEW

File: janus_core/cli/cli.py
Comment (oerc0122) [inline janus_core/cli/cli.py:1]: This still seems to have a complete rewrite of `cli.py`

Comment (oerc0122) [reply janus_core/cli/cli.py:1]: Nevermind, I see you've just broken it up. No problems here.

Comment (alinelena) [reply janus_core/cli/cli.py:1]: indeed it is...

**Review by oerc0122 — APPROVED** [review]
Looks good, much more manageable.

**Review by oerc0122 — APPROVED** [review]

**Review by alinelena — APPROVED** [review]

**Pull request merged**